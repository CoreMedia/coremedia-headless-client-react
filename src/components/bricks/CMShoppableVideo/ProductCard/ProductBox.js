// @flow
// import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import theme from 'styled-theming';

import { media } from '../../../../styles/themes/utils';

const sizeStyles = {
  tablet: theme('name', {
    default: css`
      width: 134px;
      height: 258px;
    `,
    hkm: css`
      width: 134px;
      height: 244px;
    `,
  }),
  desktop: theme('name', {
    default: css`
      width: 179px;
      height: 319px;
    `,
    hkm: css`
      width: 179px;
      height: 281px;
    `,
  }),
  large: theme('name', {
    default: css`
      width: 237px;
      height: 368px;
    `,
    hkm: css`
      width: 237px;
      height: 350px;
    `,
  }),
};

const ProductBox = styled.div`
  flex: 0 0 auto;
  position: relative;
  top: -20px;
  width: 144px;
  height: 256px;
  margin: 8px;
  z-index: 5;
  text-align: center;
  order: ${props => props.index};
  opacity: 0;
  transition: opacity 0.2s ease-out, top 0.6s ease-out;
  ${props =>
    props.entered &&
    css`
      top: 0;
      opacity: 1;
    `};
  ${props => media(props.theme.breakpoints.tablet)`
    border: 1px solid #eee;
    &:hover {
      z-index: 10;
    }
  `};
  ${props =>
    ['tablet', 'desktop', 'large'].reduce(
      (acc, breakpoint) =>
        acc.concat(
          media(props.theme.breakpoints[breakpoint])`
            ${sizeStyles[breakpoint]}
          `
        ),
      []
    )};
`;

ProductBox.displayName = 'ProductBox';

/*ProductBox.propTypes = {
  index: PropTypes.number.isRequired,
  entered: PropTypes.bool,
  theme: PropTypes.shape({
    name: PropTypes.string.isRequired,
    breakpoints: PropTypes.shape({
      tablet: PropTypes.number.isRequired,
      desktop: PropTypes.number.isRequired,
      large: PropTypes.number.isRequired,
    }).isRequired,
  }),
};*/

ProductBox.defaultProps = {
  theme: {
    name: 'default',
    breakpoints: {
      tablet: 768,
      desktop: 992,
      large: 1280,
    },
  },
};

export default ProductBox;
