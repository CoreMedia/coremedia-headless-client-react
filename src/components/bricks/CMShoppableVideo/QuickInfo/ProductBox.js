// @flow
import { Box } from '../../../basic/Box';
import { media } from '../../../../styles/themes/utils';

const ProductBox = Box.extend`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  ${props => media(props.theme.breakpoints.tablet)`
    flex-direction: row;
    align-items: start;
  `};
`;

ProductBox.displayName = 'ProductBox';

/*ProductBox.propTypes = {
  theme: PropTypes.shape({
    name: PropTypes.string.isRequired,
    breakpoints: PropTypes.shape({
      tablet: PropTypes.number.isRequired,
    }).isRequired,
  }),
};*/

ProductBox.defaultProps = {
  theme: {
    name: 'default',
    breakpoints: {
      tablet: 768,
    },
  },
};

export default ProductBox;
