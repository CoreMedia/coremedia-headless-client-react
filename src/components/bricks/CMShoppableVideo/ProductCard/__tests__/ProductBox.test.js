// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import ProductBox from '../ProductBox';

describe('ProductBox Component', () => {
  describe('with default theme', () => {
    it('should render correctly (entered=false)', () => {
      const tree = renderer
        .create(
          <ProductBox index={1} entered={false}>
            Content
          </ProductBox>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should render correctly (entered=true)', () => {
      const tree = renderer
        .create(
          <ProductBox index={1} entered={true}>
            Content
          </ProductBox>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('with hkm theme', () => {
    it('should render correctly (entered=false)', () => {
      const tree = renderer
        .create(
          <ProductBox
            index={1}
            entered={false}
            theme={{
              name: 'hkm',
              breakpoints: {
                tablet: 768,
                desktop: 992,
                large: 1280,
              },
            }}
          >
            Content
          </ProductBox>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should render correctly (entered=true)', () => {
      const tree = renderer
        .create(
          <ProductBox
            index={1}
            entered={true}
            theme={{
              name: 'hkm',
              breakpoints: {
                tablet: 768,
                desktop: 992,
                large: 1280,
              },
            }}
          >
            Content
          </ProductBox>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
