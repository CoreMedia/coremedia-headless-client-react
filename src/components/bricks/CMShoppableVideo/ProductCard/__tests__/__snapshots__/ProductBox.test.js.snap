// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductBox Component with default theme should render correctly (entered=false) 1`] = `
.c0 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  position: relative;
  top: -20px;
  width: 144px;
  height: 256px;
  margin: 8px;
  z-index: 5;
  text-align: center;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  opacity: 0;
  -webkit-transition: opacity 0.2s ease-out,top 0.6s ease-out;
  transition: opacity 0.2s ease-out,top 0.6s ease-out;
}

@media (min-width:48em) {
  .c0 {
    border: 1px solid #eee;
  }

  .c0:hover {
    z-index: 10;
  }
}

@media (min-width:48em) {
  .c0 {
    width: 134px;
    height: 258px;
  }
}

@media (min-width:62em) {
  .c0 {
    width: 179px;
    height: 319px;
  }
}

@media (min-width:80em) {
  .c0 {
    width: 237px;
    height: 368px;
  }
}

<div
  className="c0"
>
  Content
</div>
`;

exports[`ProductBox Component with default theme should render correctly (entered=true) 1`] = `
.c0 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  position: relative;
  top: -20px;
  width: 144px;
  height: 256px;
  margin: 8px;
  z-index: 5;
  text-align: center;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  opacity: 0;
  -webkit-transition: opacity 0.2s ease-out,top 0.6s ease-out;
  transition: opacity 0.2s ease-out,top 0.6s ease-out;
  top: 0;
  opacity: 1;
}

@media (min-width:48em) {
  .c0 {
    border: 1px solid #eee;
  }

  .c0:hover {
    z-index: 10;
  }
}

@media (min-width:48em) {
  .c0 {
    width: 134px;
    height: 258px;
  }
}

@media (min-width:62em) {
  .c0 {
    width: 179px;
    height: 319px;
  }
}

@media (min-width:80em) {
  .c0 {
    width: 237px;
    height: 368px;
  }
}

<div
  className="c0"
>
  Content
</div>
`;

exports[`ProductBox Component with hkm theme should render correctly (entered=false) 1`] = `
.c0 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  position: relative;
  top: -20px;
  width: 144px;
  height: 256px;
  margin: 8px;
  z-index: 5;
  text-align: center;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  opacity: 0;
  -webkit-transition: opacity 0.2s ease-out,top 0.6s ease-out;
  transition: opacity 0.2s ease-out,top 0.6s ease-out;
}

@media (min-width:48em) {
  .c0 {
    border: 1px solid #eee;
  }

  .c0:hover {
    z-index: 10;
  }
}

@media (min-width:48em) {
  .c0 {
    width: 134px;
    height: 244px;
  }
}

@media (min-width:62em) {
  .c0 {
    width: 179px;
    height: 281px;
  }
}

@media (min-width:80em) {
  .c0 {
    width: 237px;
    height: 350px;
  }
}

<div
  className="c0"
>
  Content
</div>
`;

exports[`ProductBox Component with hkm theme should render correctly (entered=true) 1`] = `
.c0 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  position: relative;
  top: -20px;
  width: 144px;
  height: 256px;
  margin: 8px;
  z-index: 5;
  text-align: center;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  opacity: 0;
  -webkit-transition: opacity 0.2s ease-out,top 0.6s ease-out;
  transition: opacity 0.2s ease-out,top 0.6s ease-out;
  top: 0;
  opacity: 1;
}

@media (min-width:48em) {
  .c0 {
    border: 1px solid #eee;
  }

  .c0:hover {
    z-index: 10;
  }
}

@media (min-width:48em) {
  .c0 {
    width: 134px;
    height: 244px;
  }
}

@media (min-width:62em) {
  .c0 {
    width: 179px;
    height: 281px;
  }
}

@media (min-width:80em) {
  .c0 {
    width: 237px;
    height: 350px;
  }
}

<div
  className="c0"
>
  Content
</div>
`;
